flow/project.py

```diff
@@ -914,25 +1233,61 @@ def _generate_id(self, job, operation_name=None, index=0):
         if max_len < len(job_op_id):
             raise ValueError("Value for MAX_LEN_ID is too small ({}).".format(self.MAX_LEN_ID))
 
+        if len(jobs) > 1:
+            aggregate_id = str(jobs[0])[:8]+'-'+str(jobs[-1])[:8]
+        else:
+            aggregate_id = str(jobs[0])[:8]
+
         separator = getattr(project._environment, 'JOB_ID_SEPARATOR', '/')
-        readable_name = '{project}{sep}{job}{sep}{op_string}{sep}{index:04d}{sep}'.format(
-                    sep=separator,
-                    project=str(project)[:12],
-                    job=str(job)[:8],
-                    op_string=op_string[:12],
-                    index=index)[:max_len]
+        readable_name = '{project}{sep}{job}{sep}{len}{sep}{group}{sep}' \
+                        '{index:04d}{sep}'.format(
+                            sep=separator,
+                            project=str(project)[:12],
+                            job=aggregate_id,
+                            len=len(jobs),
+                            group=self.name,
+                            index=index)[:max_len]
 
         # By appending the unique job_op_id, we ensure that each id is truly unique.
         return readable_name + job_op_id
 
-    def _get_status(self, job):
-        """For a given job check the groups submission status."""
+    def _get_status(self, jobs):
+        """For given jobs check the groups submission status."""
         try:
-            return JobStatus(job._project.document['_status'][self._generate_id(job)])
+            return JobStatus(jobs[0]._project.document['_status'][self._generate_id(jobs)])
```

Comment:
If jobs have changed between the submission of a group and the status check, then this will likely miss if a job is submitted for two aggregates.